other/parse_nested_brackets.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * @function parseNestedBrackets
+ * @description Parse nested brackets algorithm for a string.
+ * @param {string} w - string to parse
+ * @param { openBrackets: string; closingBrackets: string } brackets - object containing the open and closing brackets
+ * @returns {string[]} - array of the tags
+ * @example parseNestedBrackets(`<MAIN hoge><MAIN2 fuga>`) => [ '<MAIN hoge>', '<MAIN2 fuga>' ]
+ * @example parseNestedBrackets(
+ *  `THIS IS SAMPLE TEXT(MAIN hoge 0.1 fuga(ITEM fuga hoge)hoge(ITEM2 nogami(ABBR)))`,
+ *  { openBrackets: '(', closingBrackets: ')' }) => 
+ *  [
+      '(MAIN hoge 0.1 fuga(ITEM fuga hoge)hoge(ITEM2 nogami(ABBR)))',
+      '(ITEM fuga hoge)',
+      '(ITEM2 nogami(ABBR))',
+      '(ABBR)'
+    ]
+ */
+
+export const parseNestedBrackets = (
+  w: string,
+  brackets: { openBrackets: string; closingBrackets: string } = {
```

Comment:
just make it two parameters

Reply:
The characters used as brackets has no importance in this algorithm, so it should be an object in one paramter. In other words, the call like `parseNestedBrackets(v, openBrackets, closingBrackets)` looks redundant.

Reply:
The object looks even more redundant to me.

Reply:
OK. I follow your suggestion.